tus/tus-py-client
# Move to pyproject.toml and cleanup non-source files

Fixes #56, #49.

In addition to moving to `pyproject.toml` format, I've made a few other fairly significant updates:

- Add autogenerated `CONTRIBUTORS` from the git history, removed the `AUTHORS` file, and updated `LICENSE` to reference the contributors file since under the MIT Licence each contributor still maintains copyright over the code they added.
- Add more classifiers, specifically the Python versions and Typed. Changed the Development Status to Production/Stable since that probably more accurately represents that status.
- Added some badges to the `README.md`.
  - If you want to preview changes you can use a tool like [`grip`](https://github.com/joeyespo/grip)
- Removed `tox.ini`, updated `python-requires>=3.8`, added 3.12 to CI
- Moved to f-strings (removed many `"x".format(...)`)
- Applied black formatter
- Applied MyPy type checker
- Narrowed the type on `Fingerprint.get_fingerprint` - now only accepts `BinaryIO`.
  - I would be surprised if any program worked with `StringIO` files.
- removed six
- added static test files
- added .gitattributes
- lots of minor changes while on wild goose change to fix failing tests.

### Test Plan

- Check tests still pass
- Validate `pyproject.toml` against `validate-pyproject`
- Check `README.md` using `grip`
- Check project builds

### Questions / Discussion / Misc

- We may want to add a PR template to remind contributors to add their name if they haven't already.

### Note
 
I strongly recommend this is merged using a squash commit (opposed to a merge or rebase - the commit history here is pretty garbage).
